streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -177,63 +177,146 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
 
     @Override
     public void print() {
-        print(null, null, null);
+        final KeyValueMapper<? super K, ? super V, String> mapper = new KeyValueMapper<K, V, String>() {
```

Comment:
I think we can extract this mapper to a field? It is is the same in all `print` and `writeAsText` methods, right?